common/v2/features/Dashboard/components/AccountDropdown.tsx

```diff
@@ -1,136 +1,124 @@
 import React, { Component } from 'react';
 import onClickOutside from 'react-onclickoutside';
+import styled from 'styled-components';
 import { Button, Identicon } from '@mycrypto/ui';
 
+import { Checkbox } from 'v2/components';
+import { ExtendedAccount } from 'v2/services';
 import './AccountDropdown.scss';
 
-interface AccountDropdownEntry {
-  uuid: string;
-  name: string;
-  address: string;
-  visible: boolean;
+interface DropdownProps {
+  accounts: ExtendedAccount[];
+  selected: string[];
+  onSubmit(selected: string[]): void;
 }
 
-interface Props {
-  accounts: AccountDropdownEntry[];
-  visibleCount: number;
-  allVisible: boolean;
-  onSelectAll(): void;
-  onSelect(uuid: string): void;
+interface State {
+  accounts: ExtendedAccount[];
+  selected: string[];
+  open: boolean;
 }
 
-export class AccountDropdown extends Component<Props, { open: boolean }> {
-  public state = {
-    open: false
+const Divider = styled('div')`
+  border-bottom: 1px solid #e5ecf3;
+  margin-bottom: 15px;
+`;
+
+const SButton = styled(Button)`
+  height: 40px;
+  width: 100%;
+  padding: 9px 0;
+  font-size: 18px;
+  line-height: 18px;
+  font-weight: bold;
+`;
+
+class AccountDropdown extends Component<DropdownProps, State> {
+  public state: State = {
+    open: false,
+    accounts: [],
+    selected: []
   };
 
+  public componentDidMount() {
+    this.setState((_, props) => ({
+      accounts: props.accounts,
+      selected: props.selected
+    }));
+  }
+
+  // Required by onClickOutside HOC
   public handleClickOutside = () => this.state.open && this.toggleOpen();
 
+  // @TODO: TRANSLATTION_REQUIRED
   public render() {
-    const { accounts, visibleCount, allVisible, onSelectAll, onSelect } = this.props;
-    const { open } = this.state;
+    const { open, accounts, selected } = this.state;
+    const { onSubmit } = this.props;
+
+    const allVisible = accounts.length === selected.length;
     const label = allVisible
-      ? 'All Accounts'
-      : `Viewing ${visibleCount} of ${accounts.length} Accounts`;
+      ? 'All Wallets'
+      : `Viewing ${selected.length} of ${accounts.length} Accounts`;
 
     return (
-      <div role="button" onClick={this.toggleOpen} className="AccountDropdown">
+      <div
+        role="button"
+        onClick={this.toggleOpen}
+        className={`AccountDropdown ${open ? 'AccountDropdown-open' : ''}`}
```

Comment:
You can use the classnames package to make this more convenient: `className={classNames('AccountDropdown', { 'AccountDropdown-open': open })}`

Reply:
Done! Indeed it's nicer.